static/app/views/settings/organizationIntegrations/integrationServerlessFunctions.tsx

```diff
@@ -1,111 +1,97 @@
+  integration,
+}: {
   integration: OrganizationIntegration;
-  organization: Organization;
-};
-
-type State = DeprecatedAsyncComponent['state'] & {
-  serverlessFunctions: ServerlessFunction[];
-};
-
-class IntegrationServerlessFunctions extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState(): State {
-    return {
-      ...super.getDefaultState(),
-      serverlessFunctions: [],
-    };
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const orgSlug = this.props.organization.slug;
-    return [
-      [
-        'serverlessFunctions',
-        `/organizations/${orgSlug}/integrations/${this.props.integration.id}/serverless-functions/`,
-      ],
-    ];
-  }
-
-  get serverlessFunctions() {
-    return this.state.serverlessFunctions;
-  }
-
-  onLoadAllEndpointsSuccess() {
-    trackIntegrationAnalytics('integrations.serverless_functions_viewed', {
-      integration: this.props.integration.provider.key,
-      integration_type: 'first_party',
-      num_functions: this.serverlessFunctions.length,
-      organization: this.props.organization,
-    });
-  }
-
-  handleFunctionUpdate = (
-    serverlessFunctionUpdate: Partial<ServerlessFunction>,
-    index: number
-  ) => {
-    const serverlessFunctions = [...this.serverlessFunctions];
-    const serverlessFunction = {
-      ...serverlessFunctions[index],
-      ...serverlessFunctionUpdate,
-    };
-    serverlessFunctions[index] = serverlessFunction;
-    this.setState({serverlessFunctions});
-  };
-
-  renderBody() {
-    return (
-      <Fragment>
-        <Alert type="info">
-          {t(
-            'Manage your AWS Lambda functions below. Only Node and Python runtimes are currently supported.'
-          )}
-        </Alert>
-        <Panel>
-          <StyledPanelHeader disablePadding hasButtons>
-            <NameHeader>{t('Name')}</NameHeader>
-            <LayerStatusWrapper>{t('Layer Status')}</LayerStatusWrapper>
-            <EnableHeader>{t('Enabled')}</EnableHeader>
-          </StyledPanelHeader>
-          <StyledPanelBody>
-            {this.serverlessFunctions.map((serverlessFunction, i) => (
-              <IntegrationServerlessRow
-                key={serverlessFunction.name}
-                serverlessFunction={serverlessFunction}
-                onUpdateFunction={(update: Partial<ServerlessFunction>) =>
-                  this.handleFunctionUpdate(update, i)
-                }
-                {...this.props}
-              />
-            ))}
-          </StyledPanelBody>
-        </Panel>
-      </Fragment>
-    );
-  }
+}) {
+  const organization = useOrganization();
+  const queryClient = useQueryClient();
+  const queryKey: ApiQueryKey = [
+    `/organizations/${organization.slug}/integrations/${integration.id}/serverless-functions/`,
+  ];
+  const {data: serverlessFunctions = [], isSuccess} = useApiQuery<ServerlessFunction[]>(
+    queryKey,
+    {staleTime: Infinity}
```

Comment:
Might want to do `staleTime: 0` here since the list of functions can change. You'll still get the benefits of caching, it'll just refetch in the background next time you view the page.